#!/bin/sh -e
## 50_joc-toggle-buffer.dpatch by Peter S Galbraith <psg@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add joc- prefix.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad /home/psg/emacs/emacs-goodies-el/emacs-goodies-el/elisp/emacs-goodies-el/joc-toggle-buffer.el emacs-goodies-el/elisp/emacs-goodies-el/joc-toggle-buffer.el
--- /home/psg/emacs/emacs-goodies-el/emacs-goodies-el/elisp/emacs-goodies-el/joc-toggle-buffer.el	2003-10-07 19:18:29.000000000 -0400
+++ emacs-goodies-el/elisp/emacs-goodies-el/joc-toggle-buffer.el	2003-11-23 14:41:17.000000000 -0500
@@ -1,4 +1,4 @@
-;;; @(#) toggle-buffer.el --- flips back and forth between two buffers
+;;; @(#) joc-toggle-buffer.el --- flips back and forth between two buffers
 ;;; @(#) $Idx: toggle-buffer.el,v 1.1 2001/01/27 03:38:14 joe Exp $
 
 ;; Copyright (C) 2001 by Joseph L. Casadonte Jr.
@@ -63,7 +63,7 @@
 ;;  Put this file on your Emacs-Lisp load path and add the following to your
 ;;  ~/.emacs startup file
 ;;
-;;     (require 'toggle-buffer)
+;;     (require 'joc-toggle-buffer)
 
 ;;; Usage:
 ;;
@@ -100,13 +100,18 @@
 ;;  Any comments, suggestions, bug reports or upgrade requests are welcome.
 ;;  Please send them to Joe Casadonte (emacs@northbound-train.com).
 ;;
-;;  This version of toggle-buffer was developed and tested with NTEmacs 20.5.1
+;;  This version of joc-toggle-buffer was developed and tested with NTEmacs 20.5.1
 ;;  and 2.7 under Windows NT 4.0 SP6 and Emacs 20.7.1 under Linux (RH7).
 ;;  Please, let me know if it works with other OS and versions of Emacs.
 
 ;;; Change Log:
 ;;
 ;;  see http://www.northbound-train.com/emacs/toggle-buffer.log
+;;
+;;  2003-11-23 Peter S Galbraith <psg@debian.org>
+;;   This version, distributed in the Debian package `emacs-goodies-el',
+;;   was renamed from toggle-buffer.el to joc-toggle-buffer.el.  The prefix
+;;   was also added to a few variables.
 
 ;;; **************************************************************************
 ;;; **************************************************************************
@@ -152,7 +157,7 @@
   :group 'joc-toggle-buffer)
 
 ;; ---------------------------------------------------------------------------
-(defcustom toggle-buffer-load-hook nil
+(defcustom joc-toggle-buffer-load-hook nil
   "Hook to run when package is loaded."
   :type 'hook
   :group 'joc-toggle-buffer)
@@ -162,7 +167,7 @@
 ;;; **************************************************************************
 (defconst joc-toggle-buffer-version
   "$Revision: 1.1 $"
-  "Version number for toggle-buffer package.")
+  "Version number for `joc-toggle-buffer' package.")
 
 ;; ---------------------------------------------------------------------------
 (defun joc-toggle-buffer-version-number ()
@@ -182,10 +187,12 @@
 (defvar joc-toggle-buffer-last-buffer nil
   "Contains the name of the previous buffer.")
 
+;;;###autoload
 (defun joc-toggle-buffer ()
   "Switch to previous active buffer."
   (interactive)
-  (if (not (boundp 'joc-toggle-buffer-last-buffer))
+  (if (or (not (boundp 'joc-toggle-buffer-last-buffer))
+          (not joc-toggle-buffer-last-buffer))
 	  (error "No previous buffer to switch to (yet)"))
   (let ((buff (get-buffer joc-toggle-buffer-last-buffer)))
 	(if (not buff)
@@ -232,9 +239,8 @@
 ;;; **************************************************************************
 ;;; ***** we're done
 ;;; **************************************************************************
+(run-hooks 'joc-toggle-buffer-load-hook)
+
 (provide 'toggle-buffer)
-(run-hooks 'toggle-buffer-load-hook)
 
 ;;; toggle-buffer.el ends here
-;;; **************************************************************************
-;;;; *****  EOF  *****  EOF  *****  EOF  *****  EOF  *****  EOF  *************
